### PR TITLE
vim-patch:8.2.{1979,1981,1982,1985,1987}

### PR DESCRIPTION
**Pull request by janlazo**

### CODE REVIEW

File: src/nvim/edit.c [hunk 1]
```diff
@@ -8010,7 +8010,7 @@ static bool ins_bs(int c, int mode, int *inserted_space_p)
       || (!revins_on
           && ((curwin->w_cursor.lnum == 1 && curwin->w_cursor.col == 0)
               || (!can_bs(BS_START)
-                  && (arrow_used
+                  && ((arrow_used && !bt_prompt(curbuf))
```
Comment (janlazo) [inline src/nvim/edit.c:8013]: Hope that this does not need the remaining unmerged `prompt` patches.

Comment (janlazo) [reply src/nvim/edit.c]: Existing test failed so it probably does.

File: src/nvim/quickfix.c [hunk 2]
```diff
@@ -3975,11 +3975,15 @@ static void qf_fill_buffer(qf_list_T *qfl, buf_T *buf, qfline_T *old_last)
      *dirname = NUL;
 
     // Add one line for each error
-    if (old_last == NULL || old_last->qf_next == NULL) {
+    if (old_last == NULL) {
```
Comment (teto) [inline src/nvim/quickfix.c:3978]: why the refacto here ? it looks orthogonal to mbyte
